### DISCUSSION TITLE
Best practice for database calls in getServerSideProps during build?

### DISCUSSION THREAD

**Discussion by boxcee** [vercel/next.js#37587] [Help]
My code does database calls in `getServerSideProps`. For the connection to the DB however I need to have an env-variable set. Now during build (GitHub action) the env-variable is not set and therefore the build fails. What is the best practice approach here?

**Comment by Markos-Th09** [comment]
Just put the url in GitHub actions secrets and then add it as an environment variable in your action

**Reply by boxcee** [reply]
Sure, I could do that, but then I would have to expose my production environment DB to my build pipeline? Could also have a separate DB just for the pipeline, but then I would ask, why do I need to have access during build after all? Is there no other way?
